### ISSUE TITLE
Responses from /api/subreddit_stylesheet are garbled

### ISSUE THREAD

**Issue by empyrical** [reddit/reddit#1360]
Right now, if you try to make a request to `/api/subreddit_stylesheet`, you get a response that looks like this:

```js
{"jquery": [[0, 1, "call", ["#subreddit_stylesheet"]], [1, 2, "attr", "find"], [2, 3, "call", [".status"]], [3, 4, "attr", "hide"], [4, 5, "call", []], [5, 6, "attr", "html"], [6, 7, "call", [""]], [7, 8, "attr", "end"], [8, 9, "call", []], [1, 10, "attr", "find"], [10, 11, "call", [".errors"]], [11, 12, "attr", "hide"], [12, 13, "call", []], [1, 14, "attr", "find"], [14, 15, "call", [".errors ul"]], [15, 16, "attr", "hide"], ...
```

This is a response that is intended to be consumed by reddit's js, but is incredibly inconvenient and awkward to get information (like the rendered stylesheet, or if there were errors) out of it

**Comment by empyrical** [comment]
If you do that though, you'll get a very barebones response with almost no information in it

**Comment by 13steinj** [comment]
Define barebones. Care to give an example?

**Comment by empyrical** [comment]
If you supply it with the POST param `api_type: json` (.json in the url won't work for some reason) you'll get this response if there are no errors:

```json
{"json": {"errors": []}}
```

And this if there are errors:

```json
{"json": {"errors": [["BAD_CSS", "invalid css", "stylesheet_contents"]]}}
```

**Comment by 13steinj** [comment]
That's actually the *intended* response. What do you think should happen instead?

**Comment by empyrical** [comment]
The source of the rendered stylesheet if successful, and an array of the line numbers and error messages if there's an error should be provided. I can scrape them out of the `jquery` response above for now but that's not ideal, it'd be nice if this was in the official JSON response. The thing currently handling the JSON response of `subreddit_stylesheet` seems to be a generic catchall function for forms.

**Comment by 13steinj** [comment]
Gotcha. Well, I understand your concern about the error message since it is done browser side, so I'll try to make a PR for it. But the successful bit...well that's not what's supposed to happen whatsoever. If successful, an action *always* returns either `{}` or `{"errors": []}`, or a derivative of that, so that shouldn't be changed. Plus, you're editing the stylesheet, so you know what it would be in the end.

If you really want to view the stylesheet, request /about/stylesheet.json of the subreddit.

**Comment by empyrical** [comment]
The rendered stylesheet turns all the `%%image-urls%%` into their respective URLs and also is ran through the minifier, which has its own set of bugs like turning `calc(10% + 2px)` into `calc(10%+2px)` which is invalid CSS.

My usecase for this: I'm making a browser extension that has a preview tool that will allow people to browse their subreddit with a stylesheet before they save it and having the sheet exactly as reddit would end up serving it to people is important for this, imo

Here's an old PR of mine that I had to close because it became unmergable:

https://github.com/reddit/reddit/pull/1269

It's basically my "stylesheet response" wishlist. It also includes the stylesheet URL on the reddit CDN to the stylesheet when successfully saving a stylesheet.

**Comment by 13steinj** [comment]
> The rendered stylesheet turns all the %%image-urls%% into their respective URLs and also is ran through the minifier, which has its own set of bugs like turning calc(10% + 2px) into calc(10%+2px) which is invalid CSS.

No, no it doesn't that's the `/stylesheet` endpoint which is a redirect. `/about/stylesheet.json` gives the full stylesheet, uncompressed, and even contains image information and more. The invalid CSS bit you mentioned is a bug on it's own though, and is only seen on /stylesheet, not /about/stylesheet.

In my opinion (which you should keep in mind, doesn't really matter, I don't speak on behalf of reddit in any way) that for the sake of consistency and the fact that what you mentioned can already be retrieved elsewise with GET requests (which...is the standardized method to *get* data; not POST); that change shouldn't be made.

What you mentioned about the errors is definitely important though.

**Comment by empyrical** [comment]
`/api/subreddit_stylesheet` is also used for previewing the stylesheet - the preview response sends back the rendered stylesheet without saving it.

**Comment by 13steinj** [comment]
But that's a jquery based response that can't have an API JSON/XML equivalent.
